File: app/src/main/java/net/tlalka/android/fiszki/injections/components/ActivityComponent.java
package net.tlalka.android.fiszki.injections.components;

import dagger.Subcomponent;
import net.tlalka.android.fiszki.activities.HelpActivity;
import net.tlalka.android.fiszki.activities.LessonActivity;
import net.tlalka.android.fiszki.activities.LessonListActivity;
import net.tlalka.android.fiszki.activities.LessonStatsActivity;
import net.tlalka.android.fiszki.activities.MainActivity;
import net.tlalka.android.fiszki.activities.WelcomeActivity;
import net.tlalka.android.fiszki.activities.WordsActivity;
import net.tlalka.android.fiszki.injections.annotations.ActivityScope;
import net.tlalka.android.fiszki.injections.modules.ActivityModule;

@ActivityScope
@Subcomponent(modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(MainActivity mainActivity);

    void inject(WelcomeActivity welcomeActivity);

    void inject(LessonActivity lessonActivity);

    void inject(LessonListActivity lessonListActivity);

    void inject(LessonStatsActivity lessonStatsActivity);

    void inject(WordsActivity wordsActivity);

    void inject(HelpActivity helpActivity);
}

File: app/src/main/kotlin/net/tlalka/fiszki/core/components/ActivityComponent.kt
package net.tlalka.fiszki.core.components

import dagger.Subcomponent
import net.tlalka.fiszki.core.annotations.ActivityScope
import net.tlalka.fiszki.core.modules.ActivityModule
import net.tlalka.fiszki.view.activities.HelpActivity
import net.tlalka.fiszki.view.activities.LessonActivity
import net.tlalka.fiszki.view.activities.LessonListActivity
import net.tlalka.fiszki.view.activities.LessonScoreActivity
import net.tlalka.fiszki.view.activities.MainActivity
import net.tlalka.fiszki.view.activities.SettingsActivity
import net.tlalka.fiszki.view.activities.TestActivity
import net.tlalka.fiszki.view.activities.TestListActivity
import net.tlalka.fiszki.view.activities.TestScoreActivity
import net.tlalka.fiszki.view.activities.WelcomeActivity
import net.tlalka.fiszki.view.activities.WordsActivity

@ActivityScope
@Subcomponent(modules = [ActivityModule::class])
interface ActivityComponent {

    fun inject(mainActivity: MainActivity)

    fun inject(welcomeActivity: WelcomeActivity)

    fun inject(lessonActivity: LessonActivity)

    fun inject(lessonListActivity: LessonListActivity)

    fun inject(lessonScoreActivity: LessonScoreActivity)

    fun inject(testActivity: TestActivity)

    fun inject(testListActivity: TestListActivity)

    fun inject(testScoreActivity: TestScoreActivity)

    fun inject(wordsActivity: WordsActivity)

    fun inject(helpActivity: HelpActivity)

    fun inject(settingsActivity: SettingsActivity)
}
